tests/unit_tests/test_root_dir.py

```diff
@@ -0,0 +1,41 @@
+from dataregistry import DataRegistry
+import os
+import pytest
+import yaml
+
+_TEST_ROOT_DIR = "DataRegistry_data"
+_NERSC_SITE = "/global/cfs/cdirs/desc-co/registry-beta"
+_ENV_SITE = "/test/dir"
+
+
+@pytest.mark.parametrize(
+    "root_dir,site,ans",
+    [
+        (_TEST_ROOT_DIR, None, _TEST_ROOT_DIR),
+        (None, "nersc", _NERSC_SITE),
+        (None, None, _ENV_SITE),
+        (None, None, _NERSC_SITE),
+        (_TEST_ROOT_DIR, "nersc", _TEST_ROOT_DIR),
+    ],
+)
+def test_root_dir_manual(root_dir, site, ans):
+    """Test various ways of passing the root_dir"""
+
+    # Case where we are using the DATAREG_SITE env variable
+    if ans == _ENV_SITE:
+        os.environ["DATAREG_SITE"] = _ENV_SITE
+    else:
```

Comment:
You're not quite testing all cases here because the environment variable might be set when it should not be used, e.g. if root_dir is supplied.  But I don't feel strongly that every last possibility has to be tested.  Up to you whether you want to enlarge this test.

Reply:
Added a few more combinations